app/controllers/discourse_ai/admin/ai_personas_controller.rb

```diff
@@ -74,8 +74,205 @@ def destroy
         end
       end
 
+      class << self
```

Comment:
Feels like stuff that should be generic/in core/an `ActiveSupport::Concern` that we could use elsewhere too?

Reply:
I will move it out to a service at some point definitely do not want a concern here , this is a very sharp knife I only want people that absolutely understand repercussions to use it.